data/templates/sflow/hsflowd.conf.j2

```diff
@@ -0,0 +1,34 @@
+# Genereated by /usr/libexec/vyos/conf_mode/system_sflow.py
+# Parameters http://sflow.net/host-sflow-linux-config.php
+
+sflow {
+  DNSSD=off
+{% if polling is vyos_defined %}
+  polling={{ polling }}
+{% endif %}
+{% if sampling_rate is vyos_defined %}
+  # If the interface has a known ifSpeed then the samplingRate is given by the expression: ifSpeed / 1e6
+  # 100M interface will be sampled at 1-in-100
+  # 1G interface will be sampled at 1-in-1000
+  # 10G interface will be sampled at 1-in-10000
+  # 40G interface will be sampled at 1-in-40000
+  # If the ifSpeed (bits/sec) is unknown then the default given as samplingRate=N is used
```

Comment:
I did research in their documentation and here https://sflow.net/host-sflow-linux-config.php I found following:
```
sampling=400
For interfaces with no speed or applications with no specific sampling setting, fall back on this default 1-in-N rate. 
```

I *read* it as it will apply when you have no sections like following in configuration:
```
sampling.100M = 100
sampling.1G   = 500
sampling.10G  = 1000
sampling.40G  = 4000
```

I did test on my own machine with 1G interface with following setup:
```
sflow {
  DNSSD = off
  polling = 30
  agentIP = 127.0.0.1
  sampling = 150
  collector { ip = 127.0.0.1 udpport = 6343 }
  # Add all relevant interfaces to this list 
  pcap { dev = enp37s0f0 }
}
```

And I it did not behave as I expected. I have no sampling.1G here but it ignored it and used hardcoded 1:1000:
```
sudo tshark  -i lo -n -V -f 'port 6343'|grep 'Sampling rate'
Sampling rate: 1 out of 1000 packets
Sampling rate: 1 out of 1000 packets
Sampling rate: 1 out of 1000 packets
Sampling rate: 1 out of 1000 packets
Sampling rate: 1 out of 1000 packets
```

Such approach makes sampling rate setup extremely complicated and counter-intuitive. 

I believe some intermediate option like setting all speeds and fallback speed to same value will be best option:
```
  sampling.100M = {{ sampling_rate }}
  sampling.1G   = {{ sampling_rate }}
  sampling.10G  = {{ sampling_rate }}
  sampling.40G  = {{ sampling_rate }}
  sampling = {{ sampling_rate }}
```

According to our deployments 1:1000 may be safe option from performance and accuracy perspective for almost all speeds.

Reply:
Their default pre-calculated value for 10G is a way too high and it even conflicts with their official recommendations:
```
10G interface will be sampled at 1-in-10000
```